controllers/metallb_controller_test.go

```diff
@@ -122,6 +122,44 @@ var _ = Describe("MetalLB Controller", func() {
 				Expect(ok).To(BeTrue(), fmt.Sprintf("init container %s not found in %s", c.Name, speakerInitContainers))
 				Expect(c.Image).To(Equal(image))
 			}
+
+			By("Specify the speaker's Tolerations")
+			metallb.Spec.Tolerations = []v1.Toleration{
+				{
+					Key:      "example1",
+					Operator: v1.TolerationOpExists,
+					Effect:   v1.TaintEffectNoExecute,
+				},
+				{
+					Key:      "example2",
+					Operator: v1.TolerationOpExists,
+					Effect:   v1.TaintEffectNoExecute,
+				},
+			}
+
+			err = k8sClient.Update(context.TODO(), metallb)
+			Expect(err).NotTo(HaveOccurred())
+
+			speakerDaemonSet = &appsv1.DaemonSet{}
+			Eventually(func() []v1.Toleration {
+				err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: consts.MetalLBDaemonsetName, Namespace: MetalLBTestNameSpace}, speakerDaemonSet)
+				if err != nil {
+					return nil
+				}
+				return speakerDaemonSet.Spec.Template.Spec.Tolerations
+			}, 2*time.Second, 200*time.Millisecond).Should(Equal(metallb.Spec.Tolerations))
+			Expect(speakerDaemonSet).NotTo(BeZero())
+			Expect(len(speakerDaemonSet.Spec.Template.Spec.Containers)).To(BeNumerically(">", 0))
+			for _, c := range speakerDaemonSet.Spec.Template.Spec.Containers {
```

Comment:
Why do we need to check this and the initContainers?

Reply:
not need to test initconatiners as there is no speaker there removed that part